src/IO/ReadBufferFromEncryptedFile.cpp

```diff
@@ -96,6 +96,7 @@ bool ReadBufferFromEncryptedFile::nextImpl()
     working_buffer.resize(bytes_read);
     encryptor.decrypt(encrypted_buffer.data(), bytes_read, working_buffer.begin());
 
+    offset += bytes_read;
```

Comment:
How to write a test: Just insert to a table on encrypted disk any data which will be bigger than `DBMS_DEFAULT_BUFFER_SIZE` and then read all the data of this table:

```
CREATE TABLE table(number UInt64 CODEC(None)) ENGINE=MergeTree
INSERT INTO table SELECT number FROM numbers(135000)
SELECT sum(number) FROM table
```
and check the result of this select.
`CODEC(None)` should be used because we don't want compression to interfere our size calculation

Reply:
Instead of trying to make data bigger than `DBMS_DEFAULT_BUFFER_SIZE` it seems to be better to assign the setting `max_read_buffer_size` to something small, e.g. 10 bytes, while you're executing `SELECT`.